### PR TITLE
feat: add mvp jira tool bundle

### PR DESCRIPTION
**Pull request by njhale**
Add MVP Jira tools with support for:
- Issue search, create, and edit
- Listing and adding comments to an issue
- Listing projects

Addresses https://github.com/obot-platform/obot/issues/867
Supersedes https://github.com/obot-platform/tools/pull/277

**Note:** This was initially implemented in JavaScript in @tybalex's original PR, so I proceeded with that. At some point we may want to make the jump to typescript to align with the preference in this repo.

### CODE REVIEW

File: atlassian/jira/src/users.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        const users = [] 
+        for (const user of allUsers) {
+            if (includeAppUsers || user.accountType != "app") {
+                users.push({
+                    accountType: user.accountType,
+                    accountId: user.accountId,
+                    displayName: user.displayName,
+                    emailAddress: user.emailAddress,
+                    active: user.active,
+                    locale: user.locale
```
Comment (thedadams) [inline atlassian/jira/src/users.js:14]: Any reason to not include the whole user object?

File: atlassian/jira/src/users.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    try {
+        const { data: user } = await client.get(`/user?accountId=${accountId}`)
+
+        return {
+            accountType: user.accountType,
+            accountId: user.accountId,
+            displayName: user.displayName,
+            emailAddress: user.emailAddress,
+            active: user.active,
+            locale: user.locale
```
Comment (thedadams) [inline atlassian/jira/src/users.js:35]: Same question.

**Review by thedadams — APPROVED** [review]

File: atlassian/credential/tool.gpt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Name: Atlassian OAuth Credential
+Share Credential: ../../oauth2 as atlassian
+    with ATLASSIAN_OAUTH_TOKEN as env and
```
[the remark this answers is not in the document]
Comment (njhale) [reply atlassian/credential/tool.gpt:3]: Yup, saw that.

File: atlassian/jira/src/users.js (hunk 1 above)
Comment (njhale) [reply atlassian/jira/src/users.js:14]: Just to prune it down to the most important fields. See https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-users/#api-rest-api-3-user-get-response for the complete set.

File: atlassian/jira/src/users.js (hunk 2 above)
Comment (njhale) [reply atlassian/jira/src/users.js:35]: see https://github.com/obot-platform/tools/pull/291/files#r1892627717

**Review by StrongMonkey — APPROVED** [review]

**Review by drpebcak — APPROVED** [review]

**Pull request merged**